specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Subscriptions.json

```diff
@@ -0,0 +1,80 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2023-05-02-preview",
+    "title": "ContainerApps API Client"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.App/verificationId": {
```

Comment:
>"/subscriptions/{subscriptionId}/providers/Microsoft.App/verificationId" [](http://example.com/codeflow?start=4&length=72)

This API doesn't follow ARM RPC guidelines.

All the GET APIs should follow below Resource Id Contract:

If "verificationIds" is a resource type under Microsoft.App RP, below are the options:

Point/Individual GET:
GET "/subscriptions/{subscriptionId}/providers/Microsoft.App/verificationIds/{verificationIdName}"

Point/Individual GET (with Singleton Pattern):
GET "/subscriptions/{subscriptionId}/providers/Microsoft.App/verificationIds/default"

And, each Point GET should have a corresponding Collection GET:

Collection GET:
GET "/subscriptions/{subscriptionId}/providers/Microsoft.App/verificationIds"

For all the GET APIs, the response schema should be a valid ARM Resource Schema (Tracked/Proxy Resource Schema).

--------

If it is not a resource type under Microsoft.App RP, below are the options:

POST Action:
POST "/subscriptions/{subscriptionId}/providers/Microsoft.App/getVerificationId"

For the POST APIs, the response schema is up to RP.

The links to read about ARM RPC:
[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md)
[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md)
[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md)

Reply:
Changed to POST "/subscriptions/{subscriptionId}/providers/Microsoft.App/getVerificationId"